src/modm/platform/usb/sam/usb.hpp.in

```diff
@@ -0,0 +1,48 @@
+/*
+* Copyright (c) 2020, Erik Henriksson
+* Copyright (c) 2020, Niklas Hauser
+*
+* This file is part of the modm project.
+*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <modm/platform/device.hpp>
+#include <modm/platform/gpio/connector.hpp>
+
+namespace modm::platform
+{
+
+class Usb
+{
+public:
+	template< class SystemClock >
+	static void
+	initialize()
+	{
+		static_assert(SystemClock::Usb == 48_MHz, "Usb must have a 48MHz clock!");
+		PM->APBBMASK.reg |= PM_APBBMASK_USB;
+		PM->AHBMASK.reg |= PM_AHBMASK_USB;
+		GenericClockController::connect<ClockPeripheral::Usb>(ClockGenerator::System);
+	}
+
+	template< template<Peripheral _> class... Signals >
+	static void
+	connect()
+	{
+		using Connector = GpioConnector<Peripheral::Usb, Signals...>;
+		using Dp = typename Connector::template GetSignal< Gpio::Signal::Dp >;
+		using Dm = typename Connector::template GetSignal< Gpio::Signal::Dm >;
+		static_assert(((Connector::template IsValid<Dp> and Connector::template IsValid<Dm>) and sizeof...(Signals) >= 2),
```

Comment:
Should be `sizeof...(Signals) == 2`

Reply:
Ah ok, I thought perhaps you could connect more pins that just D-/D+.

Reply:
Actually, maybe you can, but in that case we should make sure I am not trying to connect the `SCK` to my sercom by passing that signal to the USB peripheral :)

I'll need to dive into the SAM family datasheet...